app/src/main/java/org/mozilla/webmaker/web/javascript/WebAppInterface.java

```diff
@@ -129,7 +128,7 @@ public String getMemStorage(String key) {
     @JavascriptInterface
     public String getMemStorage(String key, final boolean global) {
         if (!global) key = key.concat(mPrefKey);
-        return MemStorage.sharedStorage().get(key);
+        return api.getPayloads(key);
```

Comment:
since the queue system does not automatically remove queued items, and returns the list of items queued, rather than a single item, we will need to make sure to extract the element, and clear the queue:
```
String cachedItem = api.getPayload(key, 0);
api.clearPayloads(key);
return cachedItem;
```